code/modules/reagents/reagent_containers/patch.dm

```diff
@@ -17,7 +17,7 @@
 	return // thanks inheritance again
 
 /obj/item/weapon/reagent_containers/pill/patch/canconsume(mob/eater, mob/user)
-	if(!iscarbon(eater))
+	if(!iscarbon(eater) || eater.stat == DEAD)
 		return 0
 	return 1 // Masks were stopping people from "eating" patches. Thanks, inheritance.
```

Comment:
fuck you

ps. I saw no reason to re-invent the wheel with patch code when pill code did just fine